File: src/main/java/com/zerone/annotation/RequireComponent.java
package com.zerone.annotation;

import com.zerone.core.Component;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.TYPE;

@Retention(RetentionPolicy.RUNTIME)
@Target(TYPE)
public @interface RequireComponent {
    Class<? extends Component>[] value();
}

File: src/main/kotlin/com/zerone/annotation/RequireComponent.kt
package com.zerone.annotation

import com.zerone.core.Component
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class RequireComponent(vararg val value: KClass<out Component>)
